annotation-processor/src/main/java/com/hubspot/annotation/processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,112 @@
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic.Kind;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import com.google.auto.service.AutoService;
+
+@SupportedAnnotationTypes("java.lang.SuppressWarnings")
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@AutoService(Processor.class)
+public class ModernizerAnnotationProcessor extends AbstractProcessor {
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        List<String> annotatedClasses;
+        for (TypeElement annotation : annotations) {
+            annotatedClasses = getAnnotatedClasses(roundEnv, annotation);
+            makeAnnotatedClassesFile(annotatedClasses);
+        }
+        return true;
+    }
+
+    private List<String> getAnnotatedClasses(
+        RoundEnvironment roundEnv, TypeElement annotation) {
+        List<String> annotatedClasses = new ArrayList<>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings =
+                Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        FileObject fileObject;
+        Writer writer;
+        try {
+            fileObject = processingEnv.getFiler().createResource(
+                StandardLocation.CLASS_OUTPUT, "",
+                "ModernizerIgnoreAnnotatedClasses.txt", null);
+            writer = fileObject.openWriter();
+            for (String element : annotatedClasses) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            processingEnv.getMessager()
+                .printMessage(Kind.ERROR, e.getMessage());
+        }
+    }
+
+    private String getClassHeader(Element classElement) {
+        String packageName = processingEnv.getElementUtils()
+            .getPackageOf(classElement).getQualifiedName()
+            .toString().replace('.', '/');
+        Name className = classElement.getSimpleName();
+        packageName += "/" + getParentClasses(classElement) + className;
+        return packageName.replace("$", "\\$") + "(\\$.+)?";
+    }
+
+    private String getParentClasses(Element classElement) {
```

Comment:
Maybe we should call this `getFullClassName` or something and have the classElement at the end. It might simplify this for future readers who will have to know that this will be all the classes separated by and ending with `$`